# Proyecto de Analítica predictiva y modelos de regresión en Python

In [1]:
import pandas as pd

El dataset que será utilizado hace referencia a datos sobre moluscos. El objetivo de este notebook será constuir un modelo capaz de predecir la edad del molusco.

## Parte 1: Análisis exploratorio

### ¿Cuántos datos tenemos para nuestro análisis?

Importo los datos.

In [8]:
df = pd.read_csv('./data_sets/abalone_train.csv')

In [11]:
df.head()

sexo  longitud  diametro  altura  peso_entero  peso_sin_cascara  \
0    M     0.620     0.490   0.155       1.1000            0.5050   
1    I     0.475     0.375   0.110       0.4560            0.1820   
2    M     0.600     0.475   0.175       1.1100            0.5105   
3    I     0.340     0.255   0.075       0.1800            0.0745   
4    F     0.670     0.505   0.205       1.3645            0.6075   

   peso_visceras  peso_cascara  edad  
0         0.2475        0.3100     9  
1         0.0990        0.1600     9  
2         0.2560        0.2850     9  
3         0.0400        0.0525     6  
4         0.3025        0.3530     9

In [14]:
df.shape

(3342, 9)

Existen 3342 ejemplos que pueden ser utilizados para el entrenamiento del modelo. Cada ejemplo cuenta con 9 características del molusco, donde la última columna es la edad, la característica que se quiere predecir.

### Encontrar estadísticas descriptivas de todas las variables de la base de datos. 

Información general sobre los datos disponibles.

In [15]:
df.describe()

longitud     diametro       altura  peso_entero  peso_sin_cascara  \
count  3342.000000  3342.000000  3342.000000  3342.000000       3342.000000   
mean      0.522362     0.406858     0.139461     0.823557          0.355899   
std       0.120648     0.099741     0.042764     0.491525          0.221827   
min       0.075000     0.055000     0.000000     0.002000          0.001000   
25%       0.450000     0.345000     0.115000     0.438000          0.182000   
50%       0.540000     0.425000     0.140000     0.795750          0.332000   
75%       0.615000     0.480000     0.165000     1.145000          0.497000   
max       0.815000     0.650000     1.130000     2.825500          1.488000   

       peso_visceras  peso_cascara         edad  
count    3342.000000   3342.000000  3342.000000  
mean        0.179492      0.238495    10.040694  
std         0.109655      0.141137     3.335768  
min         0.000500      0.001500     1.000000  
25%         0.092500      0.127875     8.000000  
50%         0.169250      0.230000    10.000000  
75%         0.249875      0.325000    11.000000  
max         0.760000      1.005000    29.000000

### Gráficos de dispersión